### PR TITLE
[vi-mode]: Using 'dgg' to delete from the beginning of the buffer to the current logical line

### LINKED ISSUE #1749: [vi-mode] Use 'dgg' should be made to delete from the current logical line to the requested line number in a multiline buffer.
## Description of the new feature/enhancement

Bind <kbd>d</kbd><kbd>g</kbd><kbd>g</kbd> to delete from the current logical line to the requested line number in a multiline buffer.

### PR DESCRIPTION
**Pull request by springcomp**
# PR Summary

Fix #1749. 

This PR introduces a new function `DeleteRelativeLines` which deletes any number of lines before or after the current logical line in a multiline buffer.

It is used to support a new bound command <kbd>d</kbd><kbd>g</kbd><kbd>g</kbd> to delete from the beginning of the multiline buffer up to the current line.

The method is called `DeleteRelativeLines` because a numeric argument can be specified using the <kbd>d</kbd><kbd>g</kbd><kbd>g</kbd> command. This numeric argument specifies an absolute line number which, together with the current line number, make up a range of lines to be deleted. The actual number of lines to be deleted from the current logical line is computed by the difference between those two numbers, which can thus be negative. Hence the *relative* part of the name.

## PR Checklist

- [x] PR has a meaningful title
    - Use the present tense and imperative mood when describing your changes
- [x] Summarized changes
- [x] Make sure you've added one or more new tests
- **User-facing changes**
    - [x] Documentation needed at [PowerShell-Docs](https://github.com/MicrosoftDocs/PowerShell-Docs)
      - [x] Doc Issue number: [#6497](https://github.com/MicrosoftDocs/PowerShell-Docs/issues/6497)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/PowerShell/PSReadLine/pull/1752)

### CODE REVIEW

File: PSReadLine/ReadLine.vi.cs [hunk 1]
```diff
@@ -849,6 +849,28 @@ public static void DeletePreviousLines(ConsoleKeyInfo? key = null, object arg =
...
 
+        /// <summary>
+        /// Delete from the current logical line to the n-th requested logical line in a multiline buffer
+        /// </summary>
+        private static void DeleteRelativeLines(ConsoleKeyInfo? key = null, object arg = null)
+        {
+            if (TryGetArgAsInt(arg, out var requestedLineNumber, 1))
+            {
+                var currentLineIndex = _singleton.GetLogicalLineNumber() - 1;
+                var requestedLineIndex = Math.Max(requestedLineNumber, _singleton.GetStatusLineCount()) - 1;
```
Comment (daxian-dbw) [inline PSReadLine/ReadLine.vi.cs]: Why comparing `requestedLineNumber` with `_singleton.GetStatusLineCount()`? `_singleton.GetStatusLineCount()` gets you the physical line count of the `_statusLinePrompt` + `_statusBuffer`. Did you intend to use `GetLogicalLineCount`?

**Review by daxian-dbw — CHANGES_REQUESTED** [review]

**Comment by daxian-dbw** [comment]
@springcomp Could you please submit a PR to the PowerShell-doc repo for this change? You would be the best person to describe the newly added functionality. Thank you!

File: PSReadLine/ReadLine.vi.cs (hunk 1 above)
Comment (springcomp) [reply PSReadLine/ReadLine.vi.cs]: This is indeed an error.

**Comment by springcomp** [comment]
@daxian-dbw  I have addressed feedback and contributed a new PR for the documentation. Please, can you review again?

**Review by daxian-dbw — APPROVED** [review]

**Pull request merged**